src/mtype.h

```diff
@@ -268,8 +271,15 @@ class Type : public RootObject
     virtual bool isscope();
     virtual bool isString();
     virtual bool isAssignable();
+
+    virtual TypeStruct* isTypeStruct() { return NULL; }
+    virtual TypeClass* isTypeClass() { return NULL; }
+    virtual bool isTypeAggregate() { return this->isTypeStruct() || this->isTypeClass(); }
+    virtual TypeDArray* isTypeDArray() { return NULL; }
```

Comment:
Readability is subjective.  Sometimes the isXXX methods are nicer, sometimes they're not.  Adding a new redundant way to do something is not usually a good idea.  The worst of all is those changes inevitable result in discussions like this instead of useful discussion about the actual changes.

> Further there are already usages of this pattern in class Dsymbol.

Yes, `Dsymbol` uses this pattern, because `Dsymbol` doesn't have `Txxx`.  `Expression` uses `TOKxxx`.